### PR TITLE
XmlResolver: remove workaround for lack of Unix path support in Uri

### PR DESCRIPTION
**Pull request by tmds**
cc @stephentoub 

Some tests were relying on the Uri class to convert backslashes to forward slashes.

### CODE REVIEW

**Comment by stephentoub** [comment]
Thanks, @tmds.  Do we need to port this to 2.0?  Is there anything bad that happens from this workaround still being in place, or is it just a nop at this point?

**Comment by tmds** [comment]
It's worth porting due to the change in backslash conversion.

**Comment by stephentoub** [comment]
> It's worth porting due to the change in backslash conversion

That's happening in Uri because of the file:// prepended to it?

**Comment by tmds** [comment]
The LocalPath property will contain backslashes when created without the scheme. With the scheme, the backslashes will be converted to forward slashes.

**Comment by karelz** [comment]
> It's worth porting due to the change in backslash conversion.

Is that change in 2.0 already? If yes, why aren't the tests failing in 2.0? If not, why should we port it there?

**Comment by tmds** [comment]
@karelz I removed a workaround for the lack of Unix path support in the Uri class. When creating a Uri from a unix path, backslashes are now maintained instead of converted to forward slashes. e.g. new Uri("/one\two") maps to '/one\two' whereas new Uri("file:///one\two") maps to '/one/two'. Backslashes are valid in filenames in linux, so this first path is a file named 'one\two' while the other is a file named 'two' under 'one'.

**Comment by karelz** [comment]
What is the PR/issue? Did it make it into 2.0 branch?

**Comment by tmds** [comment]
In 2.0:
- workaround for lack of unix path Uri support: https://github.com/dotnet/corefx/pull/12490
- unix path Uri support: https://github.com/dotnet/corefx/pull/15925

This PR:
- undo workaround and fix tests which relied on backslash conversion happening due to workaround

**Comment by karelz** [comment]
What would be customer value of this PR in 2.0 branch?

**Comment by tmds** [comment]
This breaks existing code that has backslashes where it shouldn't. So for backwards compat reasons, better put it in 2.0.
I just did a separate PR, proposing to change the backward slash handling for all file:/// uris on Unix platforms (https://github.com/dotnet/corefx/pull/20623). If that is acceptable, we can use that one for 2.0 and leave this for 2.1.

**Comment by karelz** [comment]
@tmds I am a bit confused.
In general it would be great to first file an issue and have design / impact / compat discussion there. Discussions on PRs are troublesome.

Is this breaking compat against 1.1? If yes, then I don't see any reason to rush it into 2.0. Breaking in 2.0 or in 2.1 is basically the same, beside the general "the sooner, the better". Having it in 2.1 will also give us more bake time and opportunity to adjust based on feedback. 2.0 has near-0 runway for feedback now.

Note that 2.0 is in Escrow and pretty much locked down for high-imapctful things only.

**Comment by tmds** [comment]
@dotnet-bot test Innerloop netfx Windows_NT Release x64 Build and Test please

**Comment by sepidehMS** [comment]
@tmds I noticed you are doing some identical changes in the tests in both this PR and https://github.com/dotnet/corefx/pull/20623. I think it's better to have all Uri changes in one PR to avoid confusion, and to give a better picture of why this change is needed.

File: src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/XslTransformMultith.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public SameInstanceXslTransformTestCase(ITestOutputHelper output) : base(output)
         public new void Init(object objParam)
         {
             xsltSameInstance = new XslCompiledTransform();
-            _strPath = Path.Combine(@"TestFiles\", FilePathUtil.GetTestDataPath(), @"XsltApiV2\");
```
Comment (sepidehMS) [inline src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/XslTransformMultith.cs]: As I mentioned in another comment, since this change is dependent on the URI changes in another PR, could you please combine the PRs explaining why this change is needed? Thank you.

**Comment by tmds** [comment]
@karelz @sepidehMS This PR is removing a workaround in the XmlResolver. This is some 'cleanup'. The other PR does a more fundamental change in the Uri class. It depends on the discussion in https://github.com/dotnet/corefx/issues/20628 which way the other PR goes.

File: src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/XslTransformMultith.cs (hunk 1 above)
Comment (tmds) [inline src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/XslTransformMultith.cs]: @sepidehMS The changes in this PR do not depend on the other PR. The change in the XmlResolver to no longer prepend the file// scheme cause backslashes to be maintained. These backslashes lead to invalid paths on Unix.

**Comment by sepidehMS** [comment]
@tmds True, but still all the changes made in this PR except for the one in src/System.Private.Xml/src/System/Xml/XmlResolver.cs are already in your other PR and you will get merge conflicts if you try to merge both of them. For that reason, could you please do all this cleanup in https://github.com/dotnet/corefx/pull/20623 and close this PR?

**Comment by tmds** [comment]
@sepidehMS I will include this in the other PR when there is some progress on the related issue (https://github.com/dotnet/corefx/issues/20628).

**Comment by sepidehMS** [comment]
@tmds Thanks for doing this. Feel free to close this PR whenever you're ready.

**Comment by karelz** [comment]
Closing the PR per above discussion. @tmds please let me know if it was a bad move ...
